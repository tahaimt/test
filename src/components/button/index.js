import React,  { Component } from 'react';
import PropTypes from 'prop-types';

import withStyles from '@material-ui/styles/withStyles';
import { CssBaseline, Button, CircularProgress } from '@material-ui/core';
import { green } from '@material-ui/core/colors';

const styles = theme => ({
  button: {
    margin: theme.spacing(0.5)
  },
  buttonProgress: {
    color: green[500],
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -11,
    marginLeft: -12,
  },
  wrapper: {
    margin: theme.spacing(1),
    position: 'relative',
  }
});

export class PrimeButton extends Component {
  render() {
    const { classes } = this.props;
    return (
      <React.Fragment>
        <CssBaseline />
        <span className={classes.wrapper}>
          <Button 
            onClick={this.props.onClick}
            variant={this.props.variant}
            color={this.props.color}
            className={this.props.className}
            style={this.props.style}
            disabled={this.props.disabled || this.props.loading}
          >
            { this.props.label }
          </Button>
          {this.props.loading && <CircularProgress size={24} className={classes.buttonProgress} />}
        </span>
      </React.Fragment>
    )
  }
}

PrimeButton.propTypes = {
  onClick: PropTypes.func,
  loading: PropTypes.bool,
  variant: PropTypes.string,
  color: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  label: PropTypes.string,
  disabled: PropTypes.bool
};

export default withStyles(styles)(PrimeButton);
