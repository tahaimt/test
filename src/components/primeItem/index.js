import React,  { Component } from 'react';

import PropTypes from 'prop-types';

import withStyles from '@material-ui/styles/withStyles';
import { CssBaseline, Chip } from '@material-ui/core';

const styles = theme => ({
    wrapper: {
      margin: theme.spacing(0.5),
      position: 'relative',
    }
});

export class PrimeItem extends Component { 
    render() {
    const { classes } = this.props;

    return (
      <React.Fragment>
        <CssBaseline />
        <span className={classes.wrapper}>
          <Chip color="default" label={this.props.number}></Chip>
        </span>
      </React.Fragment>
    )
  }
}

PrimeItem.propTypes = {
  number: PropTypes.number
};

export default withStyles(styles)(PrimeItem);