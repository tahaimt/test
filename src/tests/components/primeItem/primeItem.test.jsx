// testing with react test renderer
import React from 'react';
import { PrimeItem } from "../../../components/primeItem";
import renderer from 'react-test-renderer';
import { createShallow } from "@material-ui/core/test-utils";

test('PrimeItem should match snapshot', () => {
  let number = 123;
  const component = renderer.create(
    <PrimeItem classes={{ wrapper: "container" }} number={number}></PrimeItem>,
  );
  const componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('PrimeItem should be defined', () => {
  let shallow = createShallow();
  const wrapper = shallow(
    <PrimeItem classes={{ wrapper: "wrapper" }}></PrimeItem>
  );
  expect(wrapper).toBeDefined();
});

test('PrimeItem shows correct number', () => {
  let number = 123;
  const component = renderer.create(
    <PrimeItem classes={{ wrapper: "container" }} number={number}></PrimeItem>,
  );
  const componentTree = component.toJSON();
  const textContainer = componentTree.children[0].children[0];
  expect(parseInt(textContainer.children[0])).toEqual(number);
});