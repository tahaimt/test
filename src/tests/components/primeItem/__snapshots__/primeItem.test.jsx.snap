// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeItem should match snapshot 1`] = `
<span
  className="container"
>
  <div
    className="MuiChip-root"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label"
    >
      123
    </span>
  </div>
</span>
`;
